sound/soc/sof/ipc.c

```diff
@@ -302,15 +308,13 @@ static int sof_ipc_tx_message_unlocked(struct snd_sof_ipc *ipc, u32 header,
 	/* initialise the message */
 	msg = &ipc->msg;
 
-	msg->header = header;
+	/* attach message data */
+	memcpy(msg->msg_data, msg_data, msg_bytes);
```

Comment:
any reason why you removed the if (msg_bytes) which was on line 310 before?

Reply:
We are checking the `msg_data` and only allow tx with a message, which is always the case.

We no longer need to check that there is a message to be copied as there is always a message to be copied.